Made changes to pom.xml, changelog/db.changelog-master.yaml and RepositoryConfiguration File

ShalakaSharma opened the pull request:
Signed-off-by: Shalaka <sharmask@iu.edu>

ShalakaSharma committed:
Made changes to pom.xml, changelog/db.changelog-master.yaml and RepositoryConfig.java

Signed-off-by: Shalaka <sharmask@iu.edu>

burnumd commented on pom.xml:
```diff
@@ -29,10 +29,10 @@
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-data-jpa</artifactId>
-        </dependency>
+		<dependency>
```
These changes are still using hard tabs. Please fix that in your IDE and re-indent these files.

ShalakaSharma replied:
I am not sure what do you mean by hard tabs. I am using eclipse and I generally format the code using the shortcode (Ctrll+Shift+F). Is that wrong practice for pom.xml?

burnumd commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -267,4 +251,5 @@ databaseChangeLog:
-            tableName: ACTION_REQUEST                               
\ No newline at end of file
+            tableName: ACTION_REQUEST      
+                                     
```
Remove the spaces on this last line.

ShalakaSharma replied:
Okay

burnumd commented on src/main/resources/db/changelog/db.changelog-master.yaml:
```diff
@@ -1,24 +1,14 @@
             
   - changeSet:
-      id: 2
+      id: "create sequence and table for process type"
```
Why did you decide against using UUIDs?

ShalakaSharma replied:
Because you mentioned they don't have to be numeric in one of the comments. So I decided to give logical names. Also in one of the other comments you mentioned that we can use uuids if we are using numeric ids.

ShalakaSharma replied:
If you want us to change it to UUIDs, we'll do that

burnumd replied:
Ah, that was a miscommunication. You don't need to use numeric IDs for Liquibase changesets. The data in the tables should use UUIDs for their primary keys.

ShalakaSharma replied:
Okay. So, I'll make changes in the primary keys for the tables. Remove auto increment and use UUIDS. However, there will be a problem in loadData because of this as it would be impossible to know which process_type_id to enter in the action_requests.csv file as the foreign key.

stephenpaul2727 committed:
Added Unit Tests For Converters

#issue4

Added JUnit tests for
ActionRequestedConverter,ProcessInstanceStatusConverter,RouteNodeConverter

stephenpaul2727 committed:
Fixed the Indentation Issues

Added Spaces instead of Hard Tabs.

stephenpaul2727 committed:
Removed IDE Specific Directives

ShalakaSharma committed:
Added ProcessInstance Model.
Incorporated UUIDS for primary keys
Added username for ActionRequest table
Deleted unneccessary POJOs (Link, ActionList.java, Initiator.java)
Added hateoas dependencies
Added ActionRequestResourceProcessor.java
Fixed indentation issues for pom.xml

Signed-off-by: Shalaka <sharmask@iu.edu>

ShalakaSharma committed:
Merge branch 'action_request_comments' of https://github.com/p632-sp-2017/action-list-api into action_request_comments

Conflicts:
	pom.xml
	src/main/java/com/actionlistapi/config/SecurityConfig.java
	src/main/java/com/actionlistapi/model/ActionRequest.java
	src/main/java/com/actionlistapi/model/ProcessType.java

Signed-off-by: Shalaka <sharmask@iu.edu>

burnumd commented on src/main/java/com/actionlistapi/config/ActionRequestResourceProcessor.java:
```diff
@@ -0,0 +1,43 @@
+
+    return resource;
+  }
+}
```
This needs a newline at the end of the file.

burnumd commented on src/main/java/com/actionlistapi/model/ActionRequest.java:
```diff
@@ -19,140 +19,136 @@
-	@GeneratedValue(strategy = GenerationType.IDENTITY,generator="id_seq")
-	@SequenceGenerator(name="id_seq",sequenceName="id_seq",allocationSize=1)
-	private long id;
+	@Id 
```
Remove the whitespace at the end of this line.

ShalakaSharma replied:
This is an old snippet. This as been replaced in the new commits

burnumd replied:
Oh, weird, it showed up in the "new changes since the last time you viewed."

ShalakaSharma replied:
Yeah. Really weird!

burnumd replied:
It looks like there's still a space after `@Id` in the most recent version of the code.

ShalakaSharma replied:
Yeah. Will fix it

burnumd commented on src/main/java/com/actionlistapi/model/ActionRequest.java:
```diff
@@ -19,140 +19,136 @@
+	private String username;
 	
-	private String title;
+	@OneToOne
```
Why did this relationship change to one-to-one? I'd think it would still be many-to-one.

ShalakaSharma replied:
Sure. I'll change this back to Many-to-One

burnumd commented on src/test/java/com/actionlistapi/services/ActionRequestedConverterSpec.java:
```diff
@@ -0,0 +1,43 @@
+	public void stringConvertingToActionRequestedEntityAttributeVerified() {
+		assertThat(actionRequestedConverter.convertToEntityAttribute(fakeString),instanceOf(ActionRequested.class));		
+	}
+	}
```
This closing brace is indented incorrectly.

burnumd commented on src/test/java/com/actionlistapi/services/ProcessInstanceStatusConverterSpec.java:
```diff
@@ -0,0 +1,42 @@
+import com.actionlistapi.pojo.ProcessInstanceStatus;
+
+@RunWith(SpringRunner.class)
+public class ProcessInstanceStatusConverterSpec {
```
The convention in Java is to name Unit tests as `*Test.java` (so, `ProcessInstanceStatusConverterTest` in this case)

ShalakaSharma committed:
changes for review comments
